Lib/test/test_gc.py

```diff
@@ -766,6 +766,30 @@ def test_freeze(self):
         gc.unfreeze()
         self.assertEqual(gc.get_freeze_count(), 0)
 
+    def test_get_objects(self):
+        gc.collect()
+        l = []
+        l.append(l)
+        self.assertIn(l, gc.get_objects(generation=0))
+        self.assertNotIn(l, gc.get_objects(generation=1))
+        self.assertNotIn(l, gc.get_objects(generation=2))
+        gc.collect(generation=0)
+        self.assertNotIn(l, gc.get_objects(generation=0))
+        self.assertIn(l, gc.get_objects(generation=1))
+        self.assertNotIn(l, gc.get_objects(generation=2))
+        gc.collect(generation=1)
+        self.assertNotIn(l, gc.get_objects(generation=0))
+        self.assertNotIn(l, gc.get_objects(generation=1))
+        self.assertIn(l, gc.get_objects(generation=2))
+        gc.collect(generation=2)
+        self.assertNotIn(l, gc.get_objects(generation=0))
+        self.assertNotIn(l, gc.get_objects(generation=1))
+        self.assertIn(l, gc.get_objects(generation=2))
+
```

Comment:
Please ensure also that gc.get_objects(generation=None) is valid.

Maybe test that gc.get_objects(generation=None) == gc.get_objects()?

Reply:
You're going to compare all objects in Python memory for equality?

Reply:
I will check only the lenght because checking for equality is going to be super expensive.